PowerFGT/Public/cmdb/firewall/proxy-address.ps1

```diff
@@ -0,0 +1,579 @@
+﻿#
+# Copyright 2019, Alexis La Goutte <alexis dot lagoutte at gmail dot com>
+# Copyright 2019, Benjamin Perrier <ben dot perrier at outlook dot com>
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+function Add-FGTFirewallProxyAddress {
+
+    <#
+        .SYNOPSIS
+        Add a FortiGate ProxyAddress
+
+        .DESCRIPTION
+        Add a FortiGate ProxyAddress (host-regex, url, category...)
+
+        .EXAMPLE
+        Add-FGTFirewallAddress -type ipmask -Name FGT -ip 192.2.0.0 -mask 255.255.255.0
+
+        Add Address objet type ipmask with name FGT and value 192.2.0.0/24
+
+        .EXAMPLE
+        Add-FGTFirewallAddress -type ipmask -Name FGT -ip 192.2.0.0 -mask 255.255.255.0 -interface port2
+
+        Add Address objet type ipmask with name FGT, value 192.2.0.0/24 and associated to interface port2
+
+        .EXAMPLE
+        Add-FGTFirewallAddress -type ipmask -Name FGT -ip 192.2.0.0 -mask 255.255.255.0 -comment "My FGT Address"
+
+        Add Address objet type ipmask with name FGT, value 192.2.0.0/24 and a comment
+
+        .EXAMPLE
+        Add-FGTFirewallAddress -type ipmask -Name FGT -ip 192.2.0.0 -mask 255.255.255.0 -visibility:$false
+
+        Add Address objet type ipmask with name FGT, value 192.2.0.0/24 and disabled visibility
+
+    #>
+    [CmdletBinding(PositionalBinding=$false)]
+    Param(
+        [Parameter (Mandatory = $true)]
+        [ValidateSet("host-regex", "url", "method","category",IgnoreCase = $false)]
+        [string]$type,
+        [Parameter (Mandatory = $true)]
+        [string]$name,
+        
+        [Parameter (Mandatory = $false)]
+        [ValidateLength(0, 255)]
+        [string]$comment,
+        [Parameter (Mandatory = $false)]
+        [boolean]$visibility,
+        [Parameter(Mandatory = $false)]
+        [String[]]$vdom,
+        [Parameter(Mandatory = $false)]
+        [psobject]$connection = $DefaultFGTConnection
+    )
+DynamicParam

```

Comment:
really need to use DynamicParam ?

Reply:
As discussed before, its not needed but usefull. It is also used to collect the validationset. The validationset might take up a long time when there are many objects. i only tested it on a test fortigate. Maybe not use it in large production systems.

Reply:
i plan to a next release to add better validation (with cache)
can you use ParametSet for the moment ? (like other module